test/Performance/PerformanceTestBase.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System.IO;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Microsoft.DotNet.TestFramework;
+
+namespace Microsoft.DotNet.Tools.Builder.Tests
+{
+    public class PerformanceTestBase : TestBase
+    {
+        protected void Build(string project)
```

Comment:
I think it makes more sense to inline the bodies of build and run into the individual tests.  This does two things: (1) allows you to minimize the amount of code inside of the measurement block.  (2) makes it clear what code is being executed in the measurement block.

I also considered the idea of passing the iteration object to these methods, but that means that you have much less control in the test itself of when measurement starts, and it's error prone because it's standard practice to have the call to StartMeasurement in the test method itself.

If we do that, then I think the base class probably isn't helping you much, and I'd just recommend moving the TestInstance methods into the test class itself (or if they belong in TestBase you could consider putting them there as well).